src/PythonExtensionGen.cpp

```diff
@@ -145,7 +145,11 @@ void PythonExtensionGen::compile(const Module &module) {
 extern "C" {
 #endif
 
-static __attribute__((unused)) int _convert_py_buffer_to_halide(
+static
+#if !defined(WIN32) && !defined(_WIN32)
+__attribute__((unused))
```

Comment:
Sometimes I just really hate C++. I know why this is here, but damn is it ugly (even for generated code). If we must have it, we should probably conditionalize on the compiler, not the API, i.e. `#ifndef _MSC_VER` instead of `#if !defined(WIN32) && !defined(_WIN32)`

Reply:
There is another occurrence of `#if !defined(WIN32) && !defined(_WIN32)` in the same source file.  Would you still prefer to have `#if !defined(_MSC_VER)` for this change?

Reply:
The other occurrence should go away once the explicit cast is added.

Reply:
No. there is another occurrence in the same file that is not from this PR.

```
/* Older Python versions don't set up PyMODINIT_FUNC correctly. */
#if defined(WIN32) || defined(_WIN32)
#    define HALIDE_PYTHON_EXPORT __declspec(dllexport)
#else
#    define HALIDE_PYTHON_EXPORT __attribute__((visibility("default")))
#endif
```

Reply:
Addressed it in https://github.com/halide/Halide/pull/4743